source/common/upstream/cluster_manager_impl.cc

```diff
@@ -1275,6 +1276,9 @@ void ClusterManagerImpl::postThreadLocalClusterUpdate(ClusterManagerCluster& cm_
             cluster_manager->thread_local_clusters_.size());
       }
 
+      if (cluster_manager->thread_local_clusters_[info->name()]) {
+        cluster_manager->thread_local_clusters_[info->name()]->setDropOverload(drop_overload);
```

Comment:
Do you mean add to inside the if (add_or_update_cluster) {} block?  That way the EDS update will not work since in that case by the time EDS update arrives, add_or_update_cluster is true, and the update will be skipped. If we add an else {} there and add the setDropOverload() inside it, it may have issues for static clusters.  Anyway, the idea is to keep the logic same as other fields in the below cluster_manager->updateClusterMembership().

Reply:
Sure, np! Thanks for reviewing and approval!